client/src/views/viewers/SpreadsheetViewer.vue

```diff
@@ -68,18 +80,23 @@ const currentPage = ref('');
 const loading = ref(true);
 const error = ref('');
 let worker: Worker | null = null;
-const actions: Ref<Array<{ icon?: string; text?: Translatable; handler: () => void }>> = ref([]);
 const spreadsheet = ref();
 const zoomControl = ref();
 const zoomLevel = ref(1);
+const dropdownItems = ref<FileControlsDropdownGroup>({ activeItem: 0, items: [] });
 
 onMounted(async () => {
   try {
     workbook = XLSX.read(props.contentInfo.data, { type: 'array' });
     pages.value = workbook.SheetNames;
     await switchToPage(workbook.SheetNames[0]);
+    let i = 0;
```

Comment:
You should be able to iter with `for (const [index, page] of pages.value.entries())` to get the index. Just a little tip.